vertx-db2-client/src/test/java/io/vertx/db2client/TableJoinTest.java

```diff
@@ -76,6 +79,7 @@ public void testRightOuterJoin(TestContext ctx) {
 
 	@Test
 	public void testFullOuterJoin(TestContext ctx) {
+	    assumeFalse("DB2 on Z does not support operations within ON clause for FULL OUTER JOIN", rule.isZOS());
```

Comment:
Why fail the test for Z instead of wrapping it with 
if (!rule.isZOS()) {
  testJoin(ctx, "FULL OUTER JOIN");
}